src/main/java/duck/Parser.java

```diff
@@ -0,0 +1,252 @@
+            String[] firstPart = parts[0].split("deadline ", 2);
+            String description = firstPart.length > 1 ? firstPart[1] : "";
+            String by = parts.length > 1 ? parts[1].trim() : "";
+            if (description.isBlank() || by.isBlank()) {
+                throw new EmptyDetailsException("No description or 'by' deadline time provided");
+            }
+            Task task = new Deadline(false, description, by);
+            list.add(task);
+
+            return ui.addTaskMessage(task, list);
+
+        } catch (EmptyDetailsException e) {
+            return ui.showErrorMessage(e.getMessage());
+        } catch (Exception e) {
+            return ui.showErrorMessage("Invalid date format! Use yyyy-MM-dd HHmm (e.g., 2025-01-27 2359).");
+        }
+
+    }
+
+    /**
+     * Adds an Event task to the list.
+     *
+     * @param userInput The user input containing event details and time range.
+     * @param list The task list to add the task to.
+     * @param ui The UI instance to display feedback.
+     */
+    public String event(String userInput, TaskList list, Ui ui) {
+        try {
+            String[] parts = userInput.split(" /from | /to ");
+            String[] firstPart = parts[0].split("event ", 2);
+            String description = firstPart.length > 1 ? firstPart[1] : "";
+            String from = parts.length > 1 ? parts[1].trim() : "";
+            String to = parts.length > 2 ? parts[2].trim() : "";
+            if (description.isBlank() || from.isBlank() || to.isBlank()) {
+                throw new EmptyDetailsException("No description or 'from' or 'to' time provided");
+            }
+            Task task = new Event(false, description, from, to);
+            list.add(task);
+            return ui.addTaskMessage(task, list);
+
+        } catch (EmptyDetailsException e) {
+            return ui.showErrorMessage(e.getMessage());
+        } catch (Exception e) {
+            return ui.showErrorMessage("Invalid date format! Use yyyy-MM-dd HHmm (e.g., 2025-01-27 2359).");
+        }
+    }
+
+    /**
+     * Deletes a task from the list.
+     *
+     * @param userInput The user input containing task number.
+     * @param list The task list containing the task.
+     * @param ui The UI instance to display feedback.
+     */
+    public String delete(String userInput, TaskList list, Ui ui) {
+        try {
+            String[] input = userInput.split(" ");
+            int number = Integer.parseInt(input[1]);
+            if (number <= 0 || number > list.size()) {
+                throw new IndexOutOfBoundsException("Task number out of range.");
+            }
+            Task task = list.remove(number - 1);
+            return ui.removeTaskMessage(task, list);
+        } catch (IndexOutOfBoundsException e) {
+            return ui.showErrorMessage(e.getMessage());
+        } catch (Exception e) {
+            return ui.showErrorMessage("Invalid Input, Try again!");
```

Comment:
Should specifically `RuntimeException` be caught here?
```suggestion
        } catch (RuntimeException e) {
```